commands/delete-database.go

```diff
@@ -15,81 +17,177 @@ func init() {
+			}
+		}
+
+	} else {
+		// Single database deletion
+		tgtDB := c.Args().Get(0)
+		// get all database names
+		dbs := res.MetaDB.GetDatabases()
+		if find(dbs, tgtDB) != len(dbs) {
+			names = append(names, tgtDB)
 		}
+
 	}
 
-	// delete database if it exists
-	if dbExists {
-		if res.DB.Session.DB(database).DropDatabase() != nil {
-			return cli.NewExitError("Failed to delete database", -1)
+	// check if we have databases
+	if len(names) == 0 {
+		return cli.NewExitError("Failed to find any databases", -1)
+	}
+
+	// if no force or dry run flag, verify action
+	if !force && !dryRun {
+		if confirmAction("Confirm we'll be deleting the following databases:\n" + strings.Join(names, "\n")) {
+			fmt.Println("Deleting databases...")
+		} else {
+			return cli.NewExitError("Nothing deleted, no changes have been made", 0)
 		}
 	}
 
-	// remove metadatabase record if it exists
-	if mDBExists {
-		if res.MetaDB.DeleteDB(database) != nil {
-			return cli.NewExitError("Failed to update metadb", -1)
+	// Iterate through databases to delete and delete them one by one
+	for _, database := range names {
+		dberr := deleteSingleDatabase(res, names, database, dryRun)
+		if dberr != nil {
+			return cli.NewExitError(dberr.Error, -1)
 		}
 	}
 
-	// check if we have databases
-	if err != nil || len(names) == 0 {
-		return cli.NewExitError("Failed to find any databases", -1)
+	// Dry run warning
+	if dryRun {
+		fmt.Fprintf(os.Stdout, "\t[-] This was a dry run of the delete command, nothing has been changed!\n")
+	}
+
+	return nil
+}
+
+func deleteSingleDatabase(res *resources.Resources, dbnames []string, db string, dryRun bool) error {
+	// check if database exists
+	dbExists := util.StringInSlice(db, dbnames)
+
+	// check if metadatabase record for database exists
+	mDBExists := util.StringInSlice(db, res.MetaDB.GetDatabases())
+
+	if !dryRun {
+		// delete database if it exists
+		if dbExists {
+			if res.DB.Session.DB(db).DropDatabase() != nil {
+				return errors.New("Failed to delete database")
+			}
+		}
+
+		// remove metadatabase record if it exists
+		if mDBExists {
+			if res.MetaDB.DeleteDB(db) != nil {
+				return errors.New("Failed to update metadb")
+			}
+		}
 	}
 
 	// check if we have anything to delete
 	if !dbExists && !mDBExists {
-		return cli.NewExitError("No records for database found", -1)
+		return errors.New("No records for database found")
 	}
 
 	// if it got here, deleting was a success!
-	fmt.Fprintf(os.Stdout, "\t[-] Successfully deleted database %s.\n", database)
+	fmt.Fprintf(os.Stdout, "\t[-] Successfully deleted database %s.\n", db)
 
 	return nil
 }
+
+func confirmAction(confimationMessage string) bool {
+	fmt.Print(confimationMessage, "\n [y/N] : ")
+
+	read := bufio.NewReader(os.Stdin)
+
+	response, _ := read.ReadString('\n')
+
+	response = strings.ToLower(strings.TrimSpace(response))
+	if response == "y" || response == "yes" {
+		return true
+	}
+
+	return false
+}
+
+func checkFlags(a bool, b bool, c bool) bool {
```

Comment:
It's not clear what this function does.  Can you please add a header that describes its purpose along with which cases it returns true/false?